### PR TITLE
[max] Max always update channels & avoid config updates

### PR DESCRIPTION
**Pull request by marcelrv**
Avoid OH2.5 rule triggers due to config updates
Always update channel so we always see info in the UI

### CODE REVIEW

**Comment by cweitkamp** [comment]
Can you create this PR against the "2.5.x" branch?

**Comment by cweitkamp** [comment]
One question: Should we additionally check in `MaxDevicesHandler::setDeviceConfiguration()` for `!configSet` to prevent configurations updates on the devices?

https://github.com/openhab/openhab-addons/blob/9e3ef674f532068d11c9db418b609c4d388a6ed9/bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java#L550-L554

**Comment by marcelrv** [comment]
> One question: Should we additionally check in `MaxDevicesHandler::setDeviceConfiguration()` for `!configSet` to prevent configurations updates on the devices?
> 
> https://github.com/openhab/openhab-addons/blob/9e3ef674f532068d11c9db418b609c4d388a6ed9/bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java#L550-L554

I changed it now to check if anything actually changed. If the value is already up2date, no updates.
I think that is better as no unnecessary updates are than ever done

File: bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java [hunk 1]
```diff
@@ -842,11 +834,19 @@ private void setProperties(HMessage message) {
             for (Map.Entry<String, String> entry : properties.entrySet()) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: Why are the properties not shown in PaperUI?

File: bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java [hunk 2]
```diff
@@ -500,21 +488,13 @@ public void onDeviceRemoved(MaxCubeBridgeHandler bridge, Device device) {
...
     public void onDeviceAdded(Bridge bridge, Device device) {
-        forceRefresh = true;
-    }
-
-    /**
-     * Set the forceRefresh flag to ensure update when next data is coming
-     */
-    public void setForceRefresh() {
-        forceRefresh = true;
+        //
```
Comment (J-N-K) [inline bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: remove comment

File: bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java [hunk 3]
```diff
@@ -549,18 +528,38 @@ public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
...
         try {
+            boolean config_changed = false;
             logger.debug("MAX! {} {} configuration update", device.getType(), device.getSerialNumber());
             Configuration configuration = editConfiguration();
-            configuration.put(PROPERTY_ROOMNAME, device.getRoomName());
-            configuration.put(PROPERTY_ROOMID, new BigDecimal(device.getRoomId()));
-            configuration.put(PROPERTY_DEVICENAME, device.getName());
-            configuration.put(PROPERTY_RFADDRESS, device.getRFAddress());
-            // Add additional device config entries
+            if (getConfig().get(PROPERTY_ROOMNAME) == null
```
Comment (J-N-K) [inline bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: I think the null check is not needed. `equalsIgnoreCase` can handle null arguments and returns `false` if the argument is null.

File: bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java (hunk 1 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: as this was the very first binding ported to OH2... no such thing in PaperUI at that time...
now it is... so I can indeed remove that bit of code

File: bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java (hunk 3 above)
Comment (marcelrv) [reply bundles/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: cool, did not know that. I removed the first check

**Comment by J-N-K** [comment]
Travis fails with itest infrastructure. Not related.

**Pull request merged**